### DISCUSSION TITLE
Resizing in browser produces better results than imagemagick

### DISCUSSION THREAD

**Discussion by nelio** [ImageMagick/ImageMagick#4457] [Help]
I've been using ImageMagick for many years and it always produces images with good quality, however today I've come across an image which makes ImageMagick struggle a bit. 

It's a 3840x540 PNG image that when resized to 1500x211 using the standard parameters and 80% quality produces a poor output image (very soft and some pixelated edges). However, if I simply transform it to the same image with a quality of 30% and resize it in the browser the results are noticeably better and the file sizes are similar.

I've tried different parameters (unsharpen, filters, ...) on ImageMagick but I simply can't get it to produce a proper image, so I was wondering if anyone here has encountered issue this before.

Original image:
![original](https://user-images.githubusercontent.com/232721/140818148-30adca78-2c66-41fa-b2be-1afcb17e08d1.png)

Scaled with ImageMagick:
```
magick convert -resize 1500x -quality 80 original.png w1500_q80.jpg
```
![w1500_q80](https://user-images.githubusercontent.com/232721/140818402-e187bee6-a970-4e9f-a175-77a24d076e1f.jpg)

Scaled in the browser:
```
magick convert original.png -quality 30 original_q30.jpg
```
![original_q30](https://user-images.githubusercontent.com/232721/140818487-5c94ffca-190b-4a4e-97d2-3c9db3d4a70d.jpg)

Thanks!

**Comment by snibgo** [comment]
I suggest you use "magick", not "magick convert", and put the options in their logical order:
```
magick original.png -resize 1500x -quality 80 out.jpg
```
You are resizing, and saving as lossy JPEG. Which has the greatest impact on your problem? I suspect it is the JPEG compression. Try saving as lossless, or as "-quality 100" JPEG.

> However, if I simply transform it to the same image with a quality of 30% and resize it in the browser the results are noticeably better and the file sizes are similar.

I don't understand this. If the browser resized the image and made a file size that is smaller, then why does that file have 3840x540 pixels? It hasn't resized.

**Reply by nelio** [reply]
@snibgo thanks for your reply. 
I've tried it with quality 100, and the results (albeit better) are still inferior to the image being scaled down to 1500x211 in the browser (Firefox in this case)
The browser didn't make the file size smaller, I've just opened the file that I'd converted to 30% quality (maintaing its geometry) in the browser and scaled it down so it had the same geometry as the image I'd resized using ImageMagick so I could visually compare the two.

**Reply by nelio** [reply]
@snibgo if you look at the images in my original post you can see the difference. The second image was resized by magick and the third would have been resized by your browser.

**Comment by snibgo** [comment]
You have encountered a trade-off between size and quality.

With my browser, Firefox v94.0.1, on my laptop screen (1920x1080 pixels), I can see no difference between your three images, even with a magnifying glass. They have all been resized by my browser. But you may see a difference on your computer, of course.

On my computer (with a smallish screen), resizing with IM to 1500 wide is followed by another resize, by the browser, to about 1000 wide. Compare this to your third image, which is resized only once (by the browser). Resizing twice instead of once is bound to give lower quality.

JPEG compression is designed for typical photographs, which your image isn't. You might prefer a different format such as WEBP.

**Reply by nelio** [reply]
Thanks for that. I suppose it might be OS/Screen dependent. I'm running Firefox 94.0.1 as well, and in my screen (3072 × 1920) the difference is quite noticeable when both images are at 1500x211

<img width="121" alt="Screenshot 2021-11-09 at 10 45 34" src="https://user-images.githubusercontent.com/232721/140910458-e9006bf6-7182-4c38-9664-2a407136ab00.png">
<img width="111" alt="Screenshot 2021-11-09 at 10 45 25" src="https://user-images.githubusercontent.com/232721/140910471-e54b8b52-0c91-4bff-ba03-5bb07e898501.png">

I've tried with WEBP and the results are very similar

**Comment by snibgo** [comment]
You show two images. The second is clearly soft. They are both downsampled versions of your original image, but the reduction is slight, around 75%.

You don't say how those images were made, or which you prefer. I suspect the second image has been downsampled, then upsampled.
